cpp/src/neighbors/ivf_flat/ivf_flat_interleaved_scan.cuh

```diff
@@ -216,24 +270,31 @@ struct loadAndComputeDist<kUnroll, Lambda, uint8_veclen, uint8_t, uint32_t, Comp
   {
   }
 
-  __device__ __forceinline__ void runLoadShmemCompute(const uint8_t* const& data,
+  __device__ __forceinline__ void runLoadShmemCompute(const cuvs::detail::byte_arithmetic_ptr& data,
                                                       const uint8_t* query_shared,
                                                       int loadIndex,
                                                       int shmemIndex)
   {
+    const uint8_t* data_ptr  = data.get_uint8_ptr();
+    const bool is_signed     = data.is_signed;
     constexpr int veclen_int = uint8_veclen / 4;  // converting uint8_t veclens to int
     loadIndex                = loadIndex * veclen_int;
 #pragma unroll
     for (int j = 0; j < kUnroll; ++j) {
       uint32_t encV[veclen_int];
       raft::ldg(
         encV,
-        reinterpret_cast<unsigned const*>(data) + loadIndex + j * kIndexGroupSize * veclen_int);
+        reinterpret_cast<unsigned const*>(data_ptr) + loadIndex + j * kIndexGroupSize * veclen_int);
       uint32_t queryRegs[veclen_int];
       raft::lds(queryRegs,
                 reinterpret_cast<unsigned const*>(query_shared + shmemIndex) + j * veclen_int);
 #pragma unroll
       for (int k = 0; k < veclen_int; k++) {
+        // Normalize int8_t bytes to uint8_t for Euclidean distance (translation-invariant)
+        if (is_signed) {
+          encV[k]      = normalize_int8_packed(encV[k]);
+          queryRegs[k] = normalize_int8_packed(queryRegs[k]);
+        }
         compute_dist(dist, queryRegs[k], encV[k]);
```

Comment:
Aren't we supposed to normalize only when the metric is L2Expanded?

Reply:
Yes, this code-path is only instantiated for `int8` when the metric is euclidean-based https://github.com/rapidsai/cuvs/pull/1437/files#diff-403c2fcd55246356fc13c2f369a109027c0983f6fafa31109d56f6f9cb273439R978-R980